### DISCUSSION TITLE
Can we extend this library by  integrating other libraries like amgx?

### DISCUSSION THREAD

**Discussion by reopio** [ginkgo-project/ginkgo#1699] [General]
Can we extend this library by  integrating other libraries like amgx or hypre? Like creating amgx executor in executor head file?

**Comment by upsj** [comment]
All of the libraries mentioned kind of operate at the same level of linear solvers, so it doesn't make much sense to use one on top of the other with heavy infrastructure, but we are working on some lightweight wrappers, which can be used for benchmarking.

I started some of that in https://github.com/ginkgo-project/thirdparty-benchmark-wrappers a while ago, but we will probably extend that soon.

**Reply by reopio** [reply]
Thank you, I will focus on lightweight wrappers instead of modifying the project code.
